### PR TITLE
Transformer forward

### PR DESCRIPTION
**Pull request by riveSunder**
This PR covers the forward pass for Issue #8 

Newly implemented:
* dot product self attention
* transformer encoder 
* encoder decoder attention
* transformer decoder

with supporting modifications to bijective tokenizer functions and sequence/vector parsing.

### CODE REVIEW

File: coverage.txt [hunk 1]
```diff
@@ -1,15 +1,12 @@
...
+sepia/__init__.py               0      0   100%
+sepia/common.py                 4      0   100%
+sepia/graph/__init__.py         0      0   100%
+sepia/graph/data.py            13      7    46%
+sepia/graph/functional.py      26     17    35%
+sepia/seq/__init__.py           0      0   100%
+sepia/seq/data.py              36     27    25%
+sepia/seq/functional.py        87      0   100%
+-----------------------------------------------
+TOTAL                         166     51    69%
```
Comment (riveSunder) [inline coverage.txt:12]: ✓
Note that the coverage is no long tracking the `test` code itself.

File: sepia/seq/data.py [hunk 2]
```diff
@@ -43,7 +49,7 @@ def vectors_to_sequence(sequence_vectors: jnp.array, sequence_dict: dict) -> str
 
         for key in sequence_dict.keys():
 
-            l1_distance = jnp.abs(element - sequence_dict[key])
+            l1_distance = jnp.sum(jnp.sqrt((element - sequence_dict[key])**2))
```
Comment (riveSunder) [inline sepia/seq/data.py:52]: l1_distance must be a scalar.

File: sepia/seq/data.py [hunk 3]
```diff
@@ -52,12 +58,6 @@ def vectors_to_sequence(sequence_vectors: jnp.array, sequence_dict: dict) -> str
 
         sequence += temp_character
 
-        if vectors == None:
```
Comment (riveSunder) [inline sepia/seq/data.py:55]: This was copied over from the `sequence_to_vectors` function, but there is no `vectors` in this function, `vectors_to_function` (rather it builds a string sequence to return).

File: sepia/seq/functional.py [hunk 4]
```diff
@@ -99,15 +183,14 @@ def bijective_reverse(sequence_features: jnp.array, \
     args: 
     sequence_vector represents a biological sequence (DNA, RNA, or amino acid) after being translated to a jax numpy array 
     parameters contains 3 matrices representing 3 matmuls/neural layers and optional biases for each
-    pad_to is an integer specifying the final length of the sequence. If pad_to is less than len(sequence), the sequence will be truncated
```
Comment (riveSunder) [inline sepia/seq/functional.py:102]: padding moved to `sequence_to_vectors`.

**Review by riveSunder — COMMENTED** [review]
OK to merge
